.devcontainer/Dockerfile.devcontainer

```diff
@@ -0,0 +1,66 @@
+FROM --platform=linux/amd64 ubuntu:20.04
+
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+
+# GitHub Codespaces uses /workspaces/mlflow as a working directory
+WORKDIR /workspaces/mlflow
+
+RUN apt-get update -y
+RUN apt-get install -y sudo
+RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC \
+    sudo -E apt-get install --no-install-recommends -y \
+    git curl wget jq vim tree zsh openjdk-11-jdk \
+    # Dependencies for pyenv
+    libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+
+# Install Oh My Zsh for a better shell experience. fb66b67d683935cd0e4a5282064947a28087a91b is the latest commit as of 2022-11-14.
+RUN bash -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/fb66b67d683935cd0e4a5282064947a28087a91b/tools/install.sh)"
+COPY .devcontainer/codespaces.zsh-theme /root/.oh-my-zsh/custom/themes/codespaces.zsh-theme
+ENV ZSH_THEME=codespaces
+
+# Install miniforge
+RUN sudo wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh \
+    && bash Miniforge3-Linux-x86_64.sh -b -p /root/miniconda \
+    && rm Miniforge3-Linux-x86_64.sh
+ENV PATH="/root/miniconda/bin:$PATH"
+
+# Install pyenv. v2.3.6 is the latest release as of 2022-11-14.
+RUN git clone --depth 1 --single-branch --branch v2.3.6 https://github.com/pyenv/pyenv.git /root/.pyenv
+ENV PYENV_ROOT="/root/.pyenv"
+ENV PATH="$PYENV_ROOT/bin:$PATH"
+
+# Install node and yarn
+RUN curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash - \
+    && sudo apt-get install -y nodejs \
+    && npm -g install yarn
+
+# Setup a python development environment
+ARG DEV_ENVIRONMENT_NAME="mlflow-dev-env"
+RUN conda create -y -n mlflow-dev-env python=3.8 \
+    && conda clean --all --yes \
+    && conda init zsh \
+    && echo "conda activate ${DEV_ENVIRONMENT_NAME}" >> ~/.zshrc
+
+# Install mlflow dependencies
+RUN mkdir -p mlflow
+COPY setup.py .
+COPY mlflow/__init__.py ./mlflow
+COPY mlflow/version.py ./mlflow
+COPY README.rst .
+COPY requirements ./requirements
+RUN conda run -n ${DEV_ENVIRONMENT_NAME} pip --no-cache-dir install -e .
```

Comment:
Using conda because it's faster than creating environment with pyenv + virtualenv.